packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -541,9 +540,8 @@ InternalModel.prototype = {
   },
 
   setId: function(id) {
+    Ember.assert('A record\'s id cannot be changed once it is in the loaded state', this.isNew());
```

Comment:
Looks like there are a few code paths that try to set a record with an id to the same id. Changing the assertion logic to the following should work.

```js
var oldId = this.id;
Ember.assert('A record\'s id cannot be changed once it is in the loaded state', oldId === null || id === oldId || this.isNew());
```